substrate/client/merkle-mountain-range/rpc/src/lib.rs

```diff
@@ -106,6 +106,33 @@ pub trait MmrApi<BlockHash, BlockNumber, MmrHash> {
 		at: Option<BlockHash>,
 	) -> RpcResult<LeavesProof<BlockHash>>;
 
+	/// Generate an MMR ancestry proof for the given `prev_block_number`.
+	///
+	/// This method calls into a runtime with MMR pallet included and attempts to generate
+	/// an MMR ancestry proof for the MMR root at the prior block with number `prev_block_number`,
+	/// with the reference MMR root at `best_known_block_number`. `best_known_block_number` must be
+	/// larger than the `prev_block_number` for the function to succeed.
+	///
+	/// Optionally via `at`, a block hash at which the runtime should be queried can be specified.
+	/// Optionally via `best_known_block_number`, the proof can be generated using the MMR's state
+	/// at a specific best block. Note that if `best_known_block_number` is provided, then also
+	/// specifying the block hash via `at` isn't super-useful here, unless you're generating proof
+	/// using non-finalized blocks where there are several competing forks. That's because MMR state
+	/// will be fixed to the state with `best_known_block_number`, which already points to
+	/// some historical block.
```

Comment:
why still provide `at` if not needed?

Reply:
It's not needed for the cross-chain slashing, since this either deals with equivocations against finalized blocks (or future block heights).
So I'd first removed it (https://github.com/paritytech/polkadot-sdk/pull/9295/commits/bcadda2ce67cdb472b19db722d1ea3827e5869a5), but then reverted this commit (https://github.com/paritytech/polkadot-sdk/pull/9295/commits/ac702e46df2ff9a501bffe407f59a7f53e40be7c - had a dangling TODO I just cleaned up https://github.com/paritytech/polkadot-sdk/pull/9295/commits/5ff08e7618ada5bdbf3c10434b9f76b53a52d52d) in case some other use case requires proving ancestry against a fork.
If you think it's not worth it to overload the API for this potential use case, I can also remove it entirely again.